addons/mail/models/mail_activity.py

```diff
@@ -290,8 +291,13 @@ def create(self, vals_list):
                         activity.action_notify()
 
             self.env[activity.res_model].browse(activity.res_id).message_subscribe(partner_ids=[partner_id])
-            if activity.date_deadline <= fields.Date.today():
-                self.env['bus.bus']._sendone(activity.user_id.partner_id, 'mail.activity/updated', {'activity_created': True})
+
+        todo_activities = activities.filtered(lambda act: act.date_deadline <= fields.Date.today())
+        if todo_activities:
+            self.env['bus.bus']._sendmany([
+                (activity.user_id.partner_id, 'mail.activity/updated', {'activity_created': True})
+                for activity in todo_activities
+            ])
```

Comment:
Could we accumulate "partners to notify" in the loop above?
So we avoid reading that field multiple times?

It's probably cached at that point, although not sure as an AccessError may clear the cache?